metadata/metadata.go

```diff
@@ -300,3 +300,34 @@ func (c *Client) WriteGuestAttributes(ctx context.Context, key, value string) er
 	_, err = c.httpClient.Do(req)
 	return err
 }
+
+// GetKey gets a specific metadata key.
+func (c *Client) GetKey(key string) (string, error) {
+	req, err := http.NewRequest("GET", c.metadataURL+key, nil)
+	if err != nil {
+		return "", err
+	}
+	req.Header.Add("Metadata-Flavor", "Google")
+
+	res, err := c.httpClient.Do(req)
+	if err == nil {
+		var statusCode string
+
+		switch res.StatusCode {
+		case 404:
+			statusCode = "(HTTP 404)"
+		case 412:
+			statusCode = "(HTTP 412)"
+		}
+
+		return "", fmt.Errorf("error connecting to metadata server%s: %+v", statusCode, err)
```

Comment:
do we want to explicitly log all status codes? `statusCode` would be empty here if it's anything apart from 404/412?

Reply:
Nice catch, will get it fixed. Thanks.

Reply:
Btw, we should propagate the error handling to the other places in the metadata package. I'm leaving a TODO to review it and come back to it later.

Reply:
Done.